core/Pos/Crypto/SafeSigning.hs

```diff
@@ -139,9 +139,10 @@ safeDeterministicKeyGen
 safeDeterministicKeyGen seed pp =
     bimap PublicKey (mkEncSecret pp) (safeCreateKeypairFromSeed seed pp)
 
--- | SafeSigner datatype to encapsulate sensible data
+-- | SafeSigner datatype to encapsulate sensitive data
 data SafeSigner = SafeSigner EncryptedSecretKey PassPhrase
                 | FakeSigner SecretKey
+                deriving Show
```

Comment:
what did you need it for?
it's dangerous to have `Show` instance for sensitive data like this, because it can be accidentally shown in some logs or wherever, which is bad

Reply:
Ah, never mind: `Show` instances for `EncryptedSecretKey` and `PassPhrase` are safe, it's alright.

Reply:
+1
I guess that `Show` instance is required by QuickCheck for automatic generation

Reply:
Exactly. It is required by `forAllM`.